lib/Db/FeedMapperV2.php

```diff
@@ -62,7 +65,9 @@ public function findAllFromUser(string $userId, array $params = []): array
             )
             ->where('feeds.user_id = :user_id')
             ->andWhere('feeds.deleted_at = 0')
-            ->groupBy('feeds.id')
+            ->groupBy('feeds.id', 'feeds.url', 'feeds.title', 'feeds.favicon_link', 'feeds.added',
+                      'feeds.folder_id', 'feeds.ordering', 'feeds.link', 'feeds.pinned',
+                      'feeds.update_error_count', 'feeds.last_update_error')
```

Comment:
Would this for example not cause all feeds with 0 errors to be grouped?

Reply:
So the reason I ended up with the terrible group by clause is postgres gives me an error that selected attributes need to be either in an aggregation function or listed in the group by. I couldn't think of any aggregation function to use and I'm guessing all the details of a feed should be the same for a given feed id.


Here what it looks like when I try to replicate it from the psql client. (with a bit of manual word for your reading convenience)

owncloud=> \
select oc_news_feeds.id, oc_news_feeds.url, count(oc_news_items.id) as unreadCount 
from oc_news_feeds left join oc_news_items on oc_news_items.feed_id = oc_news_feeds.id 
where oc_news_feeds.deleted_at = 0 
group by oc_news_feeds.id limit 5;
ERROR:  column "oc_news_feeds.url" must appear in the GROUP BY clause or be used in an aggregate function
LINE 1: select oc_news_feeds.id, oc_news_feeds.url, count(oc_news_it...

Reply:
I understand that postgres currently complains about the query, but I expect that if you have 2 rows:
```
id, url, update_error_count
---------------------------
1, https://example.com, 0
2, https://other-example.com, 0
```
It'll now group it by `update_error_count` and therefore only display a single row right?

Reply:
Perhaps the item count could be packed into a subquery, thereby removing the group by from the actual query?
```
select *, sub_items.cnt from feeds
left join ( select feed_id, count(unread) as cnt from items group by ... ) as sub_items
  on sub_items.feed_id = feed.id
```
_(not an actual sql query, just for explanation)_

Reply:
Ah I get what you're trying to say SMillerDev, you're worried that it'll group the the rows on the usually 0 error counts.

I guess I was assuming the group by was on the tuple of everything in the select and not grouping  on each individual term. But that's just a guess when I next get a chance I'll try setting up some simple test data and try it on the different database engines and report back.

And yes anoymouserver i think there are probably alternate ways of doing the select that might get around the ugly, long group by. But I was struggling how to do it with just plain SQL, let alone guessing how to tell the php database abstraction layer how to do it.

I think next is checking to see how the long group by actually works in the different engines. I'll try to report back in a day or two

Reply:
@detrout any update from your side?